webui/react/src/pages/Cluster.tsx

```diff
@@ -23,6 +56,70 @@ const Cluster: React.FC = () => {
 
   const availableResourceTypes = Object.keys(availableResources);
 
+  const cpuContainers = useMemo(() => {
+    const tally = {
+      running: 0,
+      total: 0,
+    };
+    resourcePools.forEach(rp => {
+      tally.total += rp.cpuContainerCapacity;
+      tally.running += rp.cpuContainersRunning;
+    });
+    return tally;
+  }, [ resourcePools ]);
+
+  const slotContainerStates = getSlotContainerStates(agents.data || []);
+
+  const getTotalGpuSlots = useCallback((resPoolName: string) => {
+    if (!agents.hasLoaded || !agents.data) return 0;
+    const resPoolAgents = agents.data.filter(agent => agent.resourcePool === resPoolName);
+    const overview = agentsToOverview(resPoolAgents);
+    return overview.GPU.total;
+  }, [ agents ]);
+
+  const columns = useMemo(() => {
+
+    const descriptionRender = (_: unknown, record: ResourcePool): React.ReactNode =>
+      <div className={css.descriptionColumn}>{record.description}</div>;
+
+    const chartRender = (_:unknown, record: ResourcePool): React.ReactNode => {
+      const containerStates: ResourceState[] =
+          getSlotContainerStates(agents.data || [], record.name);
+
+      const totalGpuSlots = getTotalGpuSlots(record.name);
+
+      if (totalGpuSlots === 0) return null;
+      return <SlotAllocationBar
+        className={css.chartColumn}
+        hideHeader
+        resourceStates={containerStates}
+        totalSlots={totalGpuSlots} />;
+
+    };
+
+    const newColumns = [ ...defaultColumns ].map(column => {
+      if (column.key === 'description') column.render = descriptionRender;
+      if (column.key === 'chart') column.render = chartRender;
+      return column;
+    });
+
+    return newColumns;
+  }, [ agents.data, getTotalGpuSlots ]);
+
+  const hideModal = useCallback(() => setRpDetail(undefined), []);
+
+  const onChange = useCallback((e: RadioChangeEvent) => {
```

Comment:
**nit:** rename to `onViewChange`?

Reply:
sounds good but leaving it to avoid changing the contents of the modules in this pr. let's rename in another PR when we run into it.